### PR TITLE
added leetcode problem 75

### PR DESCRIPTION
**Pull request by satvik202**
#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/75.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <stdlib.h>
```
Comment (Panquesito7) [inline leetcode/src/75.c:4]: I don't think libraries/headers should be added.

File: leetcode/src/75.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+        else{
+            swap(&arr[mid],&arr[end]);
+            end--;
+        }
+    }
+}
+
+
+int main(){
```
Comment (Panquesito7) [inline leetcode/src/75.c:34]: Same here.

**Review by Panquesito7 — COMMENTED** [review]
Thoughts? @TheAlgorithms/c_maintainer.

File: leetcode/src/75.c (hunk 2 above)
Comment (satvik202) [reply leetcode/src/75.c:26]: I've updated the code and removed the header files that aren't required.

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

File: leetcode/src/75.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+void sortColors(int* arr, int n){
```
Comment (tjgurwara99) [inline leetcode/src/75.c:2]:
```suggestion
#include <stdio.h>

void sortColors(int* arr, int n){
```

File: leetcode/src/75.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    void swap(int *x, int *y){
+    if (x==y)
+        return;
+   *x = *x + *y;
+    *y= *x - *y;
+    *x= *x - *y;
+}
```
Comment (tjgurwara99) [inline leetcode/src/75.c:9]: Is there a specific reason why you've defined it inside a function?

File: leetcode/src/75.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            mid++;
+            start++;
+        }
+        else{
+            swap(&arr[mid],&arr[end]);
+            end--;
+        }
+    }
+}
+int main(){
```
Comment (tjgurwara99) [inline leetcode/src/75.c:26]:
```suggestion
}

int main(){
```

Comment (tjgurwara99) [inline leetcode/src/75.c:26]: Also, I was under the impression that leetcode problems solutions in the repository shouldn't have the `main` function defined inside them...

Comment (satvik202) [reply leetcode/src/75.c]: I've removed the `main` function and made `swap` global.

File: leetcode/src/75.c [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
+void swap(int *x, int *y){
```
Comment (tjgurwara99) [inline leetcode/src/75.c:2]:
```suggestion
#include <stdio.h>

void swap(int *x, int *y){
```

File: leetcode/src/75.c [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    if (x==y)
+        return;
+   *x = *x + *y;
+    *y= *x - *y;
+    *x= *x - *y;
+}
+void sortColors(int* arr, int n){
```
Comment (tjgurwara99) [inline leetcode/src/75.c:9]:
```suggestion
}

void sortColors(int* arr, int n){
```

**Review by tjgurwara99 — APPROVED** [review]
Non-blocking minor formatting changes

File: leetcode/src/75.c [hunk 8]
```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
```
Comment (Panquesito7) [inline leetcode/src/75.c:1]: No headers should be included in LeetCode files.
```suggestion
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by tjgurwara99 — DISMISSED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**